### PR TITLE
Precursor loop fix and test

### LINKED ISSUE #116: Add test for precursor looping capability
There is currently no test for the precursor looping capability. The existing test for the general precursor functionality checks for the movement of precursors in a 2-D MSRE-like model but it does not loop the out-flowing precursors back into the core.

We can implement the test by adopting the same 2-D MSRE-like model and a separate 1-D pipe through which the precursors flow into, decay, and flow back into the core region.

### PR DESCRIPTION
**Pull request by smpark7**
Closes #116 

This PR fixes a bug in ```PrecursorAction.C``` which caused simulations with precursors looping to give an error and abort because MOOSE updated the ```direction``` variable type in MultiApp transfers from ```MooseEnum``` to ```MultiMooseEnum```.

This PR also adds a new test for precursor looping (```pre_loop```). ```pre_loop``` uses the same geometry and precursor decay data as the existing ```pre``` test. In addition to that, ```pre_loop``` uses a separate 1-D region (```sub.i```) 150cm long through which the precursors flow through and decay before introducing them back into the core. The following image compares the concentration of the longest-lived precursor group from ```pre``` and ```pre_loop``` on the left and right, respectively. The concentration in ```pre_loop``` is higher because of the precursors reentering the core.

Screenshot from 2020-07-24 02-09-57

The next image compares the concentration of the shortest-lived precursor group from ```pre``` and ```pre_loop``` on the left and right, respectively. The concentrations are the same within 4 sig.fig. precision because most of the precursors decay before they circulate back to the bottom of the core.

Screenshot from 2020-07-24 03-34-15

### CODE REVIEW

**Comment by gridley** [comment]
Good fix. This is something that really sets Moltres apart from other diffusion solvers, so it's great to see this is going to stay alive. Reviewing now...

**Comment by smpark7** [comment]
Thanks @gridley !

I'll wait abit for the others to have a look too

File: tests/pre/pre_loop.i [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+  nl_rel_tol = 1e-6
+  nl_abs_tol = 1e-5
+
+  solve_type = 'NEWTON'
+  petsc_options = '-snes_converged_reason -ksp_converged_reason -snes_linesearch_monitor'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type'
+  petsc_options_value = 'lu       NONZERO'
+  line_search = 'none'
+   # petsc_options_iname = '-snes_type'
```
Comment (katyhuff) [inline tests/pre/pre_loop.i:60]: delete line. Don't commit commented out lines which do nothing.
```suggestion
```

File: tests/pre/pre_loop.i [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  # petsc_options_value = 'test'
```
Comment (katyhuff) [inline tests/pre/pre_loop.i:61]:
```suggestion
```

File: tests/pre/pre_loop.i [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+  nl_max_its = 30
+  l_max_its = 100
+
+  picard_max_its = 5
+  picard_rel_tol = 1e-6
+  picard_abs_tol = 1e-5
+
+  dtmin = 1e-2
+  # dtmax = 1
```
Comment (katyhuff) [inline tests/pre/pre_loop.i:71]:
```suggestion
```

File: tests/pre/pre_loop.i [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  # dt = 1e-3
```
Comment (katyhuff) [inline tests/pre/pre_loop.i:72]:
```suggestion
```

**Comment by smpark7** [comment]
@katyhuff I've resolved your suggestions for removing the commented lines. Is it good to merge?

File: tests/pre/pre_loop.i [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    type = GenericMoltresMaterial
+    property_tables_root = '../../property_file_dir/newt_msre_fuel_'
+    interp_type = 'spline'
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  end_time = 500
+
```
Comment (katyhuff) [inline tests/pre/pre_loop.i:51]:
```suggestion
```

File: tests/pre/pre_loop.i [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  nl_rel_tol = 1e-6
+  nl_abs_tol = 1e-5
+
```
Comment (katyhuff) [inline tests/pre/pre_loop.i:54]:
```suggestion
```

File: tests/pre/pre_loop.i [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+  solve_type = 'NEWTON'
+  petsc_options = '-snes_converged_reason -ksp_converged_reason -snes_linesearch_monitor'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type'
+  petsc_options_value = 'lu       NONZERO'
+  line_search = 'none'
+
```
Comment (katyhuff) [inline tests/pre/pre_loop.i:60]:
```suggestion
```

File: tests/pre/pre_loop.i [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+  nl_max_its = 30
+  l_max_its = 100
+
```
Comment (katyhuff) [inline tests/pre/pre_loop.i:63]:
```suggestion
```

File: tests/pre/pre_loop.i [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+  picard_max_its = 5
+  picard_rel_tol = 1e-6
+  picard_abs_tol = 1e-5
+
```
Comment (katyhuff) [inline tests/pre/pre_loop.i:67]:
```suggestion
```

File: tests/pre/sub.i [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  end_time = 500
+
+  nl_rel_tol = 1e-6
+  nl_abs_tol = 1e-5
+
```
Comment (katyhuff) [inline tests/pre/sub.i:53]:
```suggestion
```

File: tests/pre/sub.i [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+    type = GenericMoltresMaterial
+    property_tables_root = '../../property_file_dir/newt_msre_fuel_'
+    interp_type = 'spline'
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  end_time = 500
+
```
Comment (katyhuff) [inline tests/pre/sub.i:50]:
```suggestion
```

File: tests/pre/sub.i [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+  nl_rel_tol = 1e-6
+  nl_abs_tol = 1e-5
+
+  solve_type = 'NEWTON'
+  petsc_options = '-snes_converged_reason -ksp_converged_reason -snes_linesearch_monitor'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type'
+  petsc_options_value = 'lu       NONZERO'
+  line_search = 'none'
+
```
Comment (katyhuff) [inline tests/pre/sub.i:59]:
```suggestion
```

File: tests/pre/sub.i [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+  nl_max_its = 20
+  l_max_its = 50
+
```
Comment (katyhuff) [inline tests/pre/sub.i:62]:
```suggestion
```

**Comment by smpark7** [comment]
@katyhuff Do you have any other comments for this PR?

**Pull request merged**